main.py

```diff
@@ -170,7 +180,7 @@ def is_audio_in_length_range(length):
         model=model,
         args=training_args,
         train_dataset=vectorized_datasets["train"] if training_args.do_train else None,
-        eval_dataset=vectorized_datasets["eval"] if training_args.do_eval else None,
+        eval_dataset=vectorized_datasets["val"] if training_args.do_eval and training_args.do_train else vectorized_datasets["test"] if training_args.do_eval else None,
```

Comment:
```
if training_args.do_eval and training_args.do_train:
    eval_dataset = vectorized_datasets["val"]
elif training_args.do_eval:
    eval_dataset = vectorized_datasets["test"]
else:
    eval_dataset = None
```
This might make the logic easier to follow compared to the nested ternary operator. Just a thought—feel free to keep it as is if you prefer the current style!